src/Http/Url.php

```diff
@@ -68,9 +68,8 @@ public function match($patterns)
             $patterns = [$patterns];
         }
 
-        //Remove scheme and query
-        $url = preg_replace('|(\?.*)?$|', '', (string) $this);
-        $url = preg_replace('|^(\w+://)|', '', $url);
+        //Remove scheme
+        $url = preg_replace('|^(\w+://)|', '', (string) $this);
```

Comment:
I see. I've added a bit more magic. If the pattern contains `?&`, is replaced to `[\?&]`, so you could do simply `?&utm_medium=*`